tests/test_cegar_ops_uf.cpp

```diff
@@ -1,113 +1,143 @@
-  Term x = s->make_symbol("x", sort);
-  RelationalTransitionSystem rts = counter_ts(s, x);
-  Term prop_term = rts.make_term(BVUlt, x, rts.make_term(11, sort));
-  SafetyProperty prop(s, prop_term);
-
-  shared_ptr<CegarOpsUf<IC3IA>> ceg =
-      make_shared<CegarOpsUf<IC3IA>>(prop, rts, s);
-  ceg->set_ops_to_abstract({ BVAdd });
-
-  ProverResult r = ceg->check_until(5);
+ protected:
+  void SetUp() override
+  {
+    opts.engine_ = GetParam();
+    // use cvc5 as the base solver as it supports both BV and Int
+    opts.smt_solver_ = SolverEnum::CVC5;
+    solver = create_solver(opts.smt_solver_);
+    solver->set_opt("produce-unsat-assumptions", "true");
+  }
+  PonoOptions opts;
+  SmtSolver solver;
+};
+
+TEST_P(CegOpsUfTests, BVSimpleSafe)
+{
+  if (opts.engine_ == Engine::BMC || opts.engine_ == Engine::BMC_SP) {
+    // skip BMC on safe tasks as they return UNKNOWN
+    return;
+  }
+  Sort sort = solver->make_sort(BV, 8);
+  Term x = solver->make_symbol("x", sort);
+  FunctionalTransitionSystem fts = counter_ts(solver, x);
+  Term prop_term = fts.make_term(BVUlt, x, fts.make_term(11, sort));
+  SafetyProperty prop(solver, prop_term);
+
+  shared_ptr<SafetyProver> ceg_prover = make_cegar_bv_arith_prover(
+      opts.engine_, prop, fts, solver, opts, { BVAdd });
+
+  ProverResult r = ceg_prover->check_until(5);
   ASSERT_EQ(r, ProverResult::TRUE);
 }
 
-TEST(CegOpsUf, BVSimpleUnsafe)
+TEST_P(CegOpsUfTests, BVSimpleUnsafe)
 {
-  SmtSolver s = create_solver(MSAT);
-
-  Sort sort = s->make_sort(BV, 8);
-  Term x = s->make_symbol("x", sort);
+  Sort sort = solver->make_sort(BV, 8);
+  Term x = solver->make_symbol("x", sort);
 
-  RelationalTransitionSystem rts = counter_ts(s, x);
-  Term prop_term = rts.make_term(BVUlt, x, s->make_term(10, sort));
-  SafetyProperty prop(s, prop_term);
+  FunctionalTransitionSystem fts = counter_ts(solver, x);
+  Term prop_term = fts.make_term(BVUlt, x, solver->make_term(10, sort));
+  SafetyProperty prop(solver, prop_term);
 
-  shared_ptr<CegarOpsUf<IC3IA>> ceg =
-      make_shared<CegarOpsUf<IC3IA>>(prop, rts, s);
-  ceg->set_ops_to_abstract({ BVAdd });
+  shared_ptr<SafetyProver> ceg_prover = make_cegar_bv_arith_prover(
+      opts.engine_, prop, fts, solver, opts, { BVAdd });
 
-  ProverResult r = ceg->check_until(11);
+  ProverResult r = ceg_prover->check_until(11);
   ASSERT_EQ(r, ProverResult::FALSE);
 }
 
-TEST(CegOpsUf, IntSimpleSafe)
+TEST_P(CegOpsUfTests, IntSimpleSafe)
 {
-  SmtSolver s = create_solver(MSAT);
-
-  Sort sort = s->make_sort(INT);
-  Term x = s->make_symbol("x", sort);
-  RelationalTransitionSystem rts = counter_ts(s, x);
-  Term prop_term = rts.make_term(Lt, x, rts.make_term(11, sort));
-  SafetyProperty prop(s, prop_term);
-
-  shared_ptr<CegarOpsUf<IC3IA>> ceg =
-      make_shared<CegarOpsUf<IC3IA>>(prop, rts, s);
-  ceg->set_ops_to_abstract({ Plus });
-
-  ProverResult r = ceg->check_until(5);
+  if (opts.engine_ == Engine::BMC || opts.engine_ == Engine::BMC_SP) {
```

Comment:
maybe run them and check for UNKNOWN result

Reply:
see df533eb